test-e2e/har/har.test.ts

```diff
@@ -0,0 +1,193 @@
+import 'mocha';
+import * as assert from 'assert';
+import { parseHttpArchiveText, compareEntries, isEquivalent } from './har'
+import * as fs from 'fs';
+import * as path from 'path';
+
+describe('experiment', () => {
+  it('should only care about attributes a and b', () => {
+    let model = {
+      a: {
+        b: null
+      },
+      c: null
+    }
+
+    let o1 = {
+      a: {
+        b: 2,
+        c: 3
+      },
+      c: 3,
+      d: 4
+    };
+
+    console.log(`typeof b: ${typeof(model.a.b)}`)
+
+    function pick<T>(model: T, obj: any): T {
+      return Object.keys(model).reduce((res, key) => {
+        if (!(key in obj)) return res;
+        if (model[key] === null) {
+          res[key] = obj[key]; 
+          return res;
+        }
+        res[key] = pick(model[key], obj[key]);
+        return res;
+      }, {} as T)
+    }
+    console.log(pick(model, o1))
+  })
+})
+
+describe('compareEntries', () => {
+  // First two recordings were generated on the same website using the same ajs, at different time
+  
+  const recording1 = parseHttpArchiveText(fs.readFileSync(path.join(__dirname, 'test_data', 'recording1.har'), 'utf8'));
+  const recording2 = parseHttpArchiveText(fs.readFileSync(path.join(__dirname, 'test_data', 'recording2.har'), 'utf8'));
+  // Third recording was generated on the same website as the first two, using a different AJS configuration that
+  // produces a different tracking api payload
+  const recording3 = parseHttpArchiveText(fs.readFileSync(path.join(__dirname, 'test_data', 'recording3.har'), 'utf8'));
+  
+  it('should return true for identical recordings', () => {
+    assert(compareEntries(recording1, recording1))
+  });
+
+  it('should return true for recordings generated by the same AJS on the same website', () => {
+    assert(compareEntries(recording1, recording2));
+  });
+
+  it('should return false for recordings generated by different AJS configurations on the same website', () => {
+    assert(!compareEntries(recording1, recording3),);
+  });
+});
+
+describe('isEquivalent', () => {
```

Comment:
Are these tests relevant to our E2E integration tests or are they from a boilerplate?

Reply:
These tests test the helper functions I wrote for the e2e tests.